### ISSUE TITLE
use better fonts beyond Computer Modern

### ISSUE THREAD

**Issue by hypotext** [penrose/penrose#351]
Keenan, in discussion on team mailing list, June 7:

> Yeah, I think the issue is not so much that Computer Modern is ugly (which is subjective), but that it doesn't match fonts you might want to use for TeX—see below for some examples.  Basically unless Penrose (eventually) has better font support, people will have to stick to Computer Modern in their documents or be ok with the mismatch.  This is probably ok for a lot of academics/teachers, but publishers and designers likely would want more control.

> My feeling is that the easiest way to provide this functionality is to not reinvent the wheel, and just build on top of TeX's existing font support.  This means exporting both (i) a graphics file (SVG or PDF) without typeset equations, and (ii) a TeX file that overlays text on top of the image via overpic (this is a simple little text file with some coordinates and TeX strings).  Then you can just render the whole thing in the actual TeX engine, taking advantage of all its features (typesetting, kerning, hyperlinks to symbols, etc.).  Why not.  The only technical challenge is somehow determining the bounding box, etc., for each typeset equation so that the Penrose layout engine can use it.  This is going to be the most annoying part.  Probably for final-quality diagrams for professional publishing it is ok to just do a pass where Penrose exports a bunch of .tex files, one for each equation, that individually get typeset by the TeX engine, and then we extract bounding box dimensions from the resulting files.  This is slow, but users can still use the MathJax fonts to get a rapid preview, and when they're happy with the results, can run this "offline" process.

> Not super high priority, but anyway, this is doable.  Maybe open an issue with some of this discussion (if there isn't one already?).

> Here are the examples—for each one I show what happens if you use Computer Modern math font with some other text font.  There are a number of packages that have a full set of glyphs matching a standard text font.  Here I use Palatino through \usepackage{mathpazo}, Charter through \usepackage[charter]{mathdesign}, and Garamond through \usepackage[garamond]{mathdesign}.  Whether any of these look "better" is subjective, but what's objective is that they *do* match the text:

![image](https://user-images.githubusercontent.com/2762356/84527770-82fa1000-acac-11ea-95f4-3bd33d5c7b1c.png)

**Comment by maxkrieger** [comment]
My guess is equation rendering wasn't the original emailer's concern, but rather symbol labelling/annotation in general? I think lots of textbooks use sans-serifs and bright colors when it's legible enough. Besides typographical correspondence between symbol in-diagram and symbol in-text, there isn't a big reason we need a math-renderer for non equational symbols.

**Comment by joshsunshine** [comment]
MathJax 3 only supports one font: https://docs.mathjax.org/en/latest/output/fonts.html (#653). Moving this to the bucket list (#653). Re-opening #303 to address this for print-friendly Penrose.
